test/cluster/dynamic_rackid_test.go

```diff
@@ -0,0 +1,469 @@
+								Selector: &metav1.LabelSelector{
+									MatchLabels: map[string]string{"app": "hostpath-writer"},
+								},
+								Template: v1.PodTemplateSpec{
+									ObjectMeta: metav1.ObjectMeta{
+										Labels: map[string]string{"app": "hostpath-writer"}, // must match selector
+									},
+									Spec: v1.PodSpec{
+										Containers: []v1.Container{
+											{
+												Name:    "writer",
+												Image:   "busybox",
+												Command: []string{"sh", "-c", `echo "2" > /tmp/rackid`},
+												VolumeMounts: []v1.VolumeMount{
+													{
+														Name:      "tmp-mount",
+														MountPath: "/tmp",
+													},
+												},
+												Lifecycle: &v1.Lifecycle{
+													PreStop: &v1.LifecycleHandler{
+														Exec: &v1.ExecAction{
+															Command: []string{
+																"/bin/sh", "-c", "rm -r /tmp/rackid; sleep 5",
+															},
+														},
+													},
+												},
+											},
+										},
+										Volumes: []v1.Volume{
+											{
+												Name: "tmp-mount",
+												VolumeSource: v1.VolumeSource{
+													HostPath: &v1.HostPathVolumeSource{
+														Path: "/tmp",
+														Type: &dir,
+													},
+												},
+											},
+										},
+									},
+								},
+							},
+						}
+
+						err := k8sClient.Create(ctx, ds)
+						Expect(err).NotTo(HaveOccurred())
+
+						err = waitForDaemonSetPodsRunning(
+							k8sClient, ctx, namespace, "hostpath-writer", retryInterval,
+							time.Minute*2,
+						)
+						Expect(err).NotTo(HaveOccurred())
+					},
+				)
+
+				AfterAll(
+					func() {
+						ds := &appsv1.DaemonSet{
+							ObjectMeta: metav1.ObjectMeta{
+								Name:      "hostpath-writer",
+								Namespace: namespace,
+							},
+						}
+
+						err := k8sClient.Delete(ctx, ds)
+						Expect(err).NotTo(HaveOccurred())
+					})
+
+				Context(
+					"When deploying cluster", func() {
+						It(
+							"Should deploy with dynamic rack", func() {
+
+								aeroCluster := createDummyAerospikeClusterWithHostPathVolume(clusterNamespacedName, 2, aerospikePath, "/tmp")
+								aeroCluster.Spec.RackConfig.RackIDSource = &asdbv1.RackIDSource{FilePath: aerospikePath + "/rackid"}
+								aeroCluster.Spec.RackConfig.Namespaces = []string{"test"}
+
+								Expect(DeployCluster(k8sClient, ctx, aeroCluster)).ToNot(HaveOccurred())
+
+								aeroCluster, err := getCluster(k8sClient, ctx, clusterNamespacedName)
+								Expect(err).ToNot(HaveOccurred())
+
+								podObject := &v1.Pod{}
+								Eventually(
+									func() bool {
+										err = k8sClient.Get(
+											ctx, types.NamespacedName{
+												Name:      aeroCluster.Name + "-1-0",
+												Namespace: clusterNamespacedName.Namespace,
+											}, podObject,
+										)
+
+										return isHostPathReadOnly(podObject.Status.ContainerStatuses, aerospikePath)
+									}, time.Minute, time.Second,
+								).Should(BeTrue())
```

Comment:
Do we need this check hostPathReadOnly check as we are enforcing this in the webhook.